www/annotationdb/annotationdb.js

```diff
@@ -0,0 +1,302 @@
+
+function generate_query(selected_nodes) {
+
+	// for each selected node, get Hail path of annotation
+	var selected_hail = $.map(selected_nodes, function(value, index) {
+		return value['annotation'];
+	});
+
+	// empty DOM element
+	$('span.hail-code.query').empty();
+
+	// if at least 1 annotation is selected, generate query to display
+	if (selected_hail.length) {
+
+		// split each annotation Hail path into its components
+		var split_paths = $.map(selected_hail.sort(), function(element, index){
+			return [element.split('.')];
+		});
+
+		// array of annotations to display, initialize with first annotation
+		var display = [split_paths[0]];
+
+		// picking only minimum annotations necessary to display
+		$.each(split_paths.slice(1), function(index, value){
+
+			// last annotation currently displayed
+			var last = display.slice(-1)[0];
+
+			// if current path has same beginning as last currently displayed, skip
+			if (last.toString() != value.slice(0, last.length).toString()){
+				display.push(value);
+			} 
+
+		});
+
+		// rejoin display annotation paths, combine into parameter string
+		display = "'" + $.map(display, function(element, index){
+			return element.join('.');
+		}).join("',<br>        '") + "'";
+				
+	} else {
+		// if no annotations selected, clear query panel
+		var display = '...';
+	}
+
+	// query string
+	var query =
+		'<pre>' +
+		'    .annotate_variants_db([' +
+		'<br>' +
+		"        " + display +
+		'<br>' +
+		'    ])' +
+		'</pre>';
+
+	// insert query into DOM
+	$('span.hail-code.query').html(query);
+}
+
+function select_nodes(node, selections){
+
+	var current_selections = selections;
+	if (node.hasOwnProperty('nodes')){
+		$.each(node['nodes'], function(index, child){
+			if ($.inArray(child, current_selections) == -1) {
+				current_selections.push(child);
+			}
+			current_selections = select_nodes(child, current_selections);
+		});
+	}
+
+	return current_selections;
+}
+
+function build_tree(data){
+
+	// use data dictionary to build query tree using bootstrap-tree.js library
+	$('#tree').treeview({
+		data: data,
+		multiSelect: true,
+		levels: 1,
+		expandIcon: 'glyphicon-plus',
+		collapseIcon: 'glyphicon-minus',
+		allowReselect: true,
+		preventUnselect: false,
```

Comment:
Double check these are behaving how you expect. I tried selecting gnomad.exomes and then couldn't unselect it. The CI server has a preview here: https://ci.hail.is/repository/download/HailSourceCode_HailMainline_BuildDocs/29108:id/www/index.html